google-cloud-clients/google-cloud-storage/src/main/java/com/google/cloud/storage/SignatureInfo.java

```diff
@@ -167,49 +176,80 @@ private String constructV4CanonicalRequestHash() {
         .toString();
   }
 
-  public String constructV4QueryString() {
-    ArrayListMultimap<String, String> paramMap = ArrayListMultimap.create();
+  /**
+   * Returns a TreeMap containing the user-supplied query parameters that do not have reserved keys.
+   */
+  private TreeMap<String, String> getNonReservedUserQueryParams() {
```

Comment:
Instead of returning a TreeMap would it be sufficient to return a Map<> instead?

Reply:
It didn't make sense to me to return any type of Map -- we want a Map that will iterate over the keys in sorted order, so TreeMap made the most sense here IMO.  If it were a public method, I'd be in favor of accepting whatever type of map, but this is a private method that just factors out some common functionality between the two query string construction methods.

Reply:
Sg, thanks @houglum. Let's merge it in.